include/fiction/io/write_sqd_sim_result.hpp

```diff
@@ -0,0 +1,301 @@
+        write_electron_distributions();
+
+        os << siqad::CLOSE_SIM_OUT;
+    }
+
+  private:
+    /**
+     * The simulation result to write.
+     */
+    const sidb_simulation_result<Lyt>& sim_result;
+    /**
+     * The output stream to write to.
+     */
+    std::ostream& os;
+
+    /**
+     * Converts an `std::any` to a string if it contains an alpha-numerical standard data type.
+     *
+     * @param value The value to convert.
+     * @return The string representation of the value.
+     */
+    [[nodiscard]] std::string any_to_string(const std::any& value) const noexcept
+    {
+        static const std::unordered_map<std::type_index, std::function<std::string(const std::any&)>> converters{
+            {typeid(int8_t), [](const std::any& v) { return std::to_string(std::any_cast<int8_t>(v)); }},
+            {typeid(uint8_t), [](const std::any& v) { return std::to_string(std::any_cast<uint8_t>(v)); }},
+            {typeid(int16_t), [](const std::any& v) { return std::to_string(std::any_cast<int16_t>(v)); }},
+            {typeid(uint16_t), [](const std::any& v) { return std::to_string(std::any_cast<uint16_t>(v)); }},
+            {typeid(int32_t), [](const std::any& v) { return std::to_string(std::any_cast<int32_t>(v)); }},
+            {typeid(uint32_t), [](const std::any& v) { return std::to_string(std::any_cast<uint32_t>(v)); }},
+            {typeid(int64_t), [](const std::any& v) { return std::to_string(std::any_cast<int64_t>(v)); }},
+            {typeid(uint64_t), [](const std::any& v) { return std::to_string(std::any_cast<uint64_t>(v)); }},
+            {typeid(float), [](const std::any& v) { return std::to_string(std::any_cast<float>(v)); }},
+            {typeid(double), [](const std::any& v) { return std::to_string(std::any_cast<double>(v)); }},
+            {typeid(long double), [](const std::any& v) { return std::to_string(std::any_cast<long double>(v)); }},
+            {typeid(std::string), [](const std::any& v) { return std::any_cast<std::string>(v); }},
+            {typeid(const char*), [](const std::any& v) { return std::string(std::any_cast<const char*>(v)); }},
+            {typeid(char), [](const std::any& v) { return std::string(1, std::any_cast<char>(v)); }},
+        };
+
+        const auto& type = value.type();
+        const auto& it   = converters.find(type);
+
+        return it == converters.end() ? std::string() : it->second(value);
+    }
+    /**
+     * Converts a charge distribution to a string.
+     *
+     * The string is a concatenation of the charge states of the cells in the distribution in order.
+     *
+     * The charge states are represented as follows:
+     * - NEUTRAL: '-'
+     * - NEGATIVE: '0'
+     * - POSITIVE: '1'
+     *
+     * @param surface The charge distribution to represent as a string.
+     * @return The string representation of the charge distribution.
+     */
+    std::string charge_distribution_to_string(const charge_distribution_surface<Lyt>& surface) const
```

Comment:
Do we need this function here? Can't we just use ``charge_configuration_to_string`` in ``sidb_charge_state.hpp``. To get ``std::vector<sidb_charge_state>``, we could use ``get_all_sidb_charges()``. Correct me if I am wrong

Reply:
Good point. That function unfortunately uses a different string encoding. Instead of `-`, `0`, `1` it uses `0`, `-`, `+`.

Reply:
I just saw that we need a ``+`` for positively charged DBs

Reply:
Aha! Then we can use your pointed-to function, right?

Reply:
sounds good